### PR TITLE
Dendrogram selection in 3D volume viewer

### PR DESCRIPTION
**Pull request by PennyQ**
similir to 2D image viewer

### CODE REVIEW

File: glue_vispy_viewers/volume/volume_toolbar.py [hunk 1]
```diff
@@ -8,19 +8,103 @@
...
+                              + ' value '+str(max_value)
+                # TODO: AttributeError: 'VispyVolumeViewer' object has no attribute 'show_status'
+                # self._vispy_data_viewer.show_status(status_text)
+
+                # TODO: vispy_viewer doesn't have client, how to get dendrogram data?
+                print('visible_data, client.data', self.visible_data, self._vispy_widget.client.data)
+                for each_data in self._vispy_widget.client.data:
+                    if each_data.label == 'Dendrogram':
+                        d = each_data
+
```
Comment (PennyQ) [inline glue_vispy_viewers/volume/volume_toolbar.py]: @astrofrog # TODO: vispy_widget doesn't have a client, how to get dendrogram data here?

**Comment by PennyQ** [comment]
@astrofrog Do you know any solutions to get `self._vispy_widget.client.data` here, so I could implement 3d selection based on dendrogram structure? Thanks!

File: glue_vispy_viewers/volume/volume_toolbar.py (hunk 1 above)
Comment (astrofrog) [reply glue_vispy_viewers/volume/volume_toolbar.py]: I think you can try `self.parent().session.data_collection` - any luck?

**Comment by PennyQ** [comment]
@astrofrog So I fix this pr which allows you to select a dendrogram structure from 3D volume viewer, one question is this requires 'astrodendro' package and better with dendrogram viewer available, shall we add them to glue-depts? Thanks ;)

**Comment by PennyQ** [comment]
new in https://github.com/glue-viz/glue-vispy-viewers/pull/233

**Pull request closed without merging**